backend/director/agents/meme_maker.py

```diff
@@ -0,0 +1,269 @@
+        except Exception:
+            self.output_message.actions.append(
+                "Transcript unavailable. Indexing spoken content."
+            )
+            self.output_message.push_update()
+            self.videodb_tool.index_spoken_words(video_id)
+            return self.videodb_tool.get_transcript(
+                video_id
+            ), self.videodb_tool.get_transcript(video_id, text=False)
+
+    def run(
+        self, prompt: str, video_id: str, collection_id: str, *args, **kwargs
+    ) -> AgentResponse:
+        """
+        Run the agent to generate meme clips and images based on user prompts.
+
+        :return: The response containing information about the sample processing operation.
+        :rtype: AgentResponse
+        """
+        try:
+            self.videodb_tool = VideoDBTool(collection_id=collection_id)
+            try:
+                _, transcript = self._get_transcript(video_id=video_id)
+                scene_index_id, scenes = self._get_scenes(video_id=video_id)
+
+                self.output_message.actions.append("Identifying meme content..")
+                self.output_message.push_update()
+                result = self._multimodal_prompter(transcript, scenes, prompt)
+
+            except Exception as e:
+                logger.exception(f"Error in getting video content: {e}")
+                return AgentResponse(status=AgentStatus.ERROR, message=str(e))
+
+            if not result:
+                return AgentResponse(
+                    status=AgentStatus.SUCCESS,
+                    message="No meme content found.",
+                    data={},
+                )
+            self.output_message.actions.append("Key moments identified..")
+            self.output_message.actions.append("Creating video clips..")
+            self.output_message.push_update()
+            success_data = {"clip_timestamps": [], "image_timestamps": []}
+            all_clips_generated: bool = True
+            for clip in result["clip_timestamps"]:
+                video_content = VideoContent(
+                    agent_name=self.agent_name,
+                    status=MsgStatus.progress,
+                    status_message="Generating clip..",
+                )
+                self.output_message.content.append(video_content)
+                try:
+                    stream_url = stream_url = self.videodb_tool.generate_video_stream(
+                        video_id=video_id,
+                        timeline=[(clip["start"], clip["end"])],
+                    )
+                except Exception as e:
+                    video_content.status_message = f"Error generating clip: {str(e)}"
+                    video_content.status = MsgStatus.error
+                    self.output_message.push_update()
+                    clip["stream_url"] = f"Error generating stream: {str(e)}"
+                    success_data["clip_timestamps"].append(clip)
```

Comment:
Can we put rename it to `data` instead of `success_data` and in case of error put the message in another list?
It should signal the reasoning engine failed as well as keep the success data seperately.

Reply:
Renamed it to data. And added error field, not creating another list for error since timestamps are correct.